#issue  24 This is my example of the function

jancris100 opened the pull request.

jancris100 commented:
@Birdmannn

Birdmannn commented:
> @Birdmannn

On it now

Birdmannn commented on poker-texas-hold-em/contract/src/systems/actions.cairo:
```diff
@@ -236,9 +237,56 @@ pub mod actions {
+            world.write_model(@current_dealer);
+
+            // Set the next dealer to is_dealer
+            let mut next_dealer: Player = world.read_model(next_dealer_address);
```
Here, I haven't implemented an optimized scenario is make a player `leave` a game. So currently, when a player is no longer in the game, the most optimized way is to leave the player in the array, perhaps the player may join back in the next two rounds or so, if the game_params permits it... and when the number of players that leave has reached a certain threshold, then the players array can be resolved in the game.


At the beginning of the function, I said something about if the player is in the game. I think it should be checked here... but the player can be in a game, and not in a round, so we will only do one check here.

When you extracted the dealer successfully earlier, is because it's safe that a player that's not in the game/round can never be a dealer, so it wasn't checked. Now this dealer assignment in this check makes your previous dealer extraction safe as it is....


before assigning the next dealer, check if the player is in round `player.in_round`. If not, move to the next index. This should probably be done in a loop, until one of the players is `in_round`. If the index finally goes through without finding any player's in the round and comes back to the current_index (`current_index == next_index`) it should panic.


so it should be in a loop. break when the next dealer is in game, and save the next dealer, or panic (`assert(next_dealer_index != current_dealer_index, 'ONLY ONE PLAYER IN GAME')`).


👀 That's why you're a Doctor, sir 😋

Birdmannn commented on poker-texas-hold-em/contract/src/systems/actions.cairo:
```diff
@@ -236,9 +237,56 @@ pub mod actions {
+            world.write_model(@next_dealer);
+
+            // Return the next dealer
+            Option::Some(next_dealer)
```
And yes 🙌🙌♠️🔥 this return is perfect

Birdmannn reviewed (commented):
Everything looking almost hale and hearty. I loved that your code is good. Tres Bien!!! Doctorr 🎉🎉. 

Justa few minor stitches, we clip the code from further unknown infections/complications. So far the work is done good, with no further complications, this task will be set to leave the hospital.


Thank you sooo much 🙏🏻🙇🏻

Birdmannn commented:
And please @jancris100, apply scarb fit when you're done so that the workflow turns green 🟢

jancris100 commented:
@Birdmannn Okay, I opened another PR for you, accidentally, both have the updated function, I removed the repeated lines and added the player and a loop just as you said to check if the player is there or not, I don't know how to run the scarb fmt from here, I ran the other one before uploading, see which PR works better and we close the other, both are the same, the new one has the scarb fmt

Birdmannn commented:
A very great job @jancris100 🫠💥. I hope we work together some time in the future 😇❄️
